### DISCUSSION TITLE
Decouple dipole and particle orientation

### DISCUSSION THREAD

**Discussion by dav10a** [espressomd/espresso#4496] [Q&A]
Hi,

I'm doing a project where I combine espresso with a micromagnetic simulation. Quickly summarized, I have a cube consisting of 64 particles, forced together with vs_auto_relate. My virtual reference particle has a dipole moment and interacts with a homogeneous magnetic field. In each time step, I am also computing  a full micromagnetic model of the cube, which means the the dipole can rotate "freely" inside my body. What I'd like to do then is to set this updated dipole to be the dipole of my espresso particle. Unfornately, this physically rotates the particle, since the dipole moment is always forced to point in the z direction, aligning with the director. Is there a way to decouple the dipole from the orientation of the particle?

**Comment by RudolfWeeber** [comment]
If you just need to manually set the orientation (and thus the dople moment) ov virtual sites, you can do so by means of the vs_quat attribute. It controls the orientation of the virtual site relative to the orientation of the base particle.

https://espressomd.github.io/doc/particles.html#virtual-sites

 

 

If you need to control dipole moment and orientation of the base particle independently, place a virtual site with distance 0 relative to the base particle and give it the dipole moment.

 

If you need the rotational degrees of freedom of virtual sites to be integrated according to an equation of motion, you need to apply a small patch to Espresso. This will likely become more convenient in a future version, where we will give more fine-grained control over the integration.
